woocommerce/woocommerce
# Only hide .payment_box elements on init for multiple payment methods

Prevents hiding a single payment method’s UI on the order pay page.

Some of our hosted gateways will output a payment form using the `woocommerce_receipt_{payment_method}` action on the order pay page, and surround it with a `.payment_box` div to get the usual styling. The changes introduced with https://github.com/woocommerce/woocommerce/commit/9cb0d3c88480623fbe0ed769d8f96db40c2e2d4f will hide _all_ `div.payment_box` elements on that page, and single the markup added might not include the payment method radio, there's no way to expand it. 

### All Submissions:


### Changes proposed in this Pull Request:

Only slideUp `.payment_box` elements if there is more than one payment method on the page.

An alternative would be to skip `init_payment_methods()` on the receipt/order pay page. 

### How to test the changes in this Pull Request:

1. Visit an order's pay page using the URL generated by `WC_Order::get_checkout_payment_url( true );`
2. Add some a `.payment_box` element using the `woocommerce_receipt_{payment_method}` action
3. Without this PR, the payment box is hidden on page load and cannot be shown
4. With the PR, it is shown by default since it is the only one that exists

### Other information:
